dev/AppLifecycle/AppInstance.cpp

```diff
@@ -308,51 +316,67 @@ namespace winrt::Microsoft::Windows::AppLifecycle::implementation
-            if (args)
+            if (auto args = winrt::Windows::ApplicationModel::AppInstance::GetActivatedEventArgs())
             {
-                data = args.as<IInspectable>();
+                data = args;
                 kind = static_cast<ExtendedActivationKind>(args.Kind());
+                foundArgs = true;
             }
         }
-        else
+
+        // Handle all Reunion types next (both packaged and unpackaged).
+        std::wstring commandLine = std::wstring(GetCommandLine());
+        if (!foundArgs)
         {
-            // Generate IActivatedEventArgs for non-Packaged applications.
-            std::wstring contractId;
-            std::wstring contractData;
-            auto commandLine = std::wstring(GetCommandLine());
-            std::tie(contractId, contractData) = ParseCommandLine(commandLine);
+            auto [contractArgument, contractData] = ParseCommandLine(commandLine);
 
             // All specific launch types are encoded as a URI and transported as a
             // protocol, except the catch-all LaunchActivatedEventArgs case.
-            if (!contractId.empty() && (CompareStringOrdinal(contractId.c_str(), -1, c_protocolArgumentString, -1, TRUE) == CSTR_EQUAL))
+            if (!contractArgument.empty())
             {
-                kind = ExtendedActivationKind::Protocol;
-                auto args = make<ProtocolActivatedEventArgs>(contractData);
-                data = args.as<IInspectable>();
-
-                // Encoded launch is a protocol launch where the argument data is
-                // encapsulated in the Uri Query data.  We handle that here and
-                // try to return the correct IActivatedEventArgs type that is
-                // encoded in the data rather than the IProtocolActivatedEventArgs
-                // itself.
-                auto protocolArgs = args.as<IProtocolActivatedEventArgs>();
-                if (IsEncodedLaunch(protocolArgs.Uri()))
+                if (contractData.empty())
                 {
-                    std::tie(kind, data) = GetEncodedLaunchActivatedEventArgs(protocolArgs);
+                    // If the contractData is empty, handle any aliased encoded launches.
+                    if (CompareStringOrdinal(contractArgument.data(), static_cast<int>(contractArgument.size()), L"ReunionPushServer", -1, TRUE) == CSTR_EQUAL)
+                    {
+                        contractData = GenerateEncodedLaunchUri(L"App", c_pushContractId);
+                        contractArgument = c_protocolArgumentString;
+                    }
                 }
-            }
 
-            // Haven't set the kind yet, and so let's default to Launch.
-            if (kind == ExtendedActivationKind::Launch)
-            {
-                data = make<LaunchActivatedEventArgs>(commandLine).as<IInspectable>();
+                if (CompareStringOrdinal(contractArgument.c_str(), -1, c_protocolArgumentString, -1, TRUE) == CSTR_EQUAL)
```

Comment:
>contractArgument [](http://example.com/codeflow?start=41&length=16)

`contractArgument.data(), static_cast<int>(contractArgument.size())` like we did to check for "ReunionPushServer".